### PR TITLE
Handled error attachments

### PR DESCRIPTION
**Pull request by guperrot**
[AB#71072](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/71072)

### CODE REVIEW

**Comment by guperrot** [comment]
Tested manually on android without committing app changes.

**Review by owlstronaut — DISMISSED** [review]

**Review by owlstronaut — DISMISSED** [review]

File: Assets/AppCenter/Plugins/iOS/Core/Utility/NSStringDictionaryHelper.mm [hunk 1]
```diff
@@ -22,8 +22,12 @@
 
 NSDictionary* appcenter_unity_create_ns_string_dictionary(char** keys, char** values, int count)
 {
+  if (count == 0) {
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/iOS/Core/Utility/NSStringDictionaryHelper.mm:22]: Let's format this file

File: Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.h [hunk 2]
```diff
@@ -4,11 +4,12 @@
 #ifndef CRASHES_UNITY_H
 #define CRASHES_UNITY_H
 
+#import <Foundation/Foundation.h>
 #import <AppCenterCrashes/AppCenterCrashes.h>
+#import "MSException.h"
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.h]: No need to import it here. Forward declaration should be enough

File: Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.h [hunk 3]
```diff
@@ -20,8 +21,15 @@ extern "C" void* appcenter_unity_crashes_last_session_crash_report();
...
+extern "C" void* app_center_unity_crashes_create_error_attachments_array(int capacity);
+extern "C" void* app_center_unity_crashes_create_error_attachment_log_text(char* text, char* fileName);
+extern "C" void* app_center_unity_crashes_create_error_attachment_log_binary(const void* data, int size, char* fileName, char* contentType);
+extern "C" void appcenter_unity_crashes_add_error_attachment(NSMutableArray<MSErrorAttachmentLog*>* attachments, MSErrorAttachmentLog* attachment);
+
+@interface MSCrashes ()
+
++ (void)trackModelException:(MSException *)exception withProperties:(nullable NSDictionary<NSString *, NSString *> *)properties withAttachments:(nullable NSArray<MSErrorAttachmentLog *> *)attachments;
+
+@end
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.h]: I guess we don't need this in header file

File: Assets/AppCenter/Plugins/AppCenterSDK/Crashes/iOS/NativeObjectsConverter.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+{
+    public static class NativeObjectsConverter
+    {
+        public static IntPtr ToNativeAttachments(ErrorAttachmentLog[] logs)
+        {
+            if (logs == null)
+            {
+                return IntPtr.Zero;
+            }
+            IntPtr nativeArray = app_center_unity_crashes_create_error_attachments_array(logs.Length);
```
Comment (MatkovIvan) [inline Assets/AppCenter/Plugins/AppCenterSDK/Crashes/iOS/NativeObjectsConverter.cs]:
```suggestion
            var nativeArray = app_center_unity_crashes_create_error_attachments_array(logs.Length);
```

File: Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.h (hunk 3 above)
Comment (guperrot) [reply Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.h]: Why not?

File: Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.h (hunk 2 above)
Comment (guperrot) [reply Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.h]: @jaelim-ms is that typical in our ios Code to just forward declare and not include?

File: Assets/AppCenter/Plugins/iOS/Core/Utility/NSStringDictionaryHelper.mm (hunk 1 above)
Comment (guperrot) [reply Assets/AppCenter/Plugins/iOS/Core/Utility/NSStringDictionaryHelper.mm:22]: The LLVM plugin that we used in XCode UI is buggy. I had to copy the file in apple repo, format it there and bring it back...

File: Assets/AppCenter/Plugins/AppCenterSDK/Crashes/iOS/NativeObjectsConverter.cs (hunk 4 above)
Comment (guperrot) [reply Assets/AppCenter/Plugins/AppCenterSDK/Crashes/iOS/NativeObjectsConverter.cs]: Will push as part of other changes.

**Review by owlstronaut — DISMISSED** [review]

**Review by MatkovIvan — DISMISSED** [review]

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**